[POS Orders] Enable `pointOfSaleOrdersi1`

iamgabrielma opened the pull request:
## Description
This PR enables the feature flag `pointOfSaleOrdersi1`, launching POS Orders i1

## Testing information
- On a store with POS and WooCommerce 9.9+, process an order through POS. 
- Upon navigating to the orders tab, observe the `POS` badge appears both in order list and order details for those orders.

| Tablet | Order list | Order details |
|--------|--------|--------|
| ![Simulator Screenshot - iPad mini (A17 Pro) - US store - 2025-07-09 at 15 28 19](https://github.com/user-attachments/assets/dce7ea0f-54fb-4b26-b6f4-45efa0e8c42d) | ![Simulator Screenshot - iPhone 16 Plus - US store - 2025-07-09 at 15 29 56](https://github.com/user-attachments/assets/eaa43f72-5f93-466d-8d6c-adbe17ec8e74) | ![Simulator Screenshot - iPhone 16 Plus - US store - 2025-07-09 at 15 30 00](https://github.com/user-attachments/assets/fb04e224-8cf1-45ee-942c-87d18cd86a1f) | 

There are some accessibility improvements to be done as part of WOOMOB-751, but these are non-blocking for feature release and will be handled as part of backlog.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

iamgabrielma committed:
enable pointOfSaleOrdersi1

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/250">22.8</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

iamgabrielma committed:
update release notes

staskus reviewed (approved):
✅  I'm approving this.

❌  However, one thing that is not working properly is the "POS" badge is not reacting to dark/light mode changes. I need to kill and relaunch the app for the colors to change.

https://github.com/user-attachments/assets/8965926d-84f8-4c48-bd75-caa8c950b0b3

iamgabrielma committed:
update badge color init so dynamic appearance is correctly applied

iamgabrielma commented:
Thanks for the review!

> ❌ However, one thing that is not working properly is the "POS" badge is not reacting to dark/light mode changes. I need to kill and relaunch the app for the colors to change.

Oh that's a great catch, appreciated.  

So it would seem that if we initialize UIColors with `UIColor(color: ...` , we're creating static colors that don't respond to trait collection changes as they're unchanged for the whole app lifecycle, we have to initialize them  dynamically instead.

Fixed on: https://github.com/woocommerce/woocommerce-ios/pull/15887/commits/42c98a9b54d34e15789499714bbb102e5a9f8d24

https://github.com/user-attachments/assets/6e332cb7-fd7d-4144-a749-abcf1aa2bf10

_PD: I don't think the dark background is a great choice when we have a selected cell in iPad, I'll raise this with design_

iamgabrielma committed:
update badge color init so does not change on appearance switch

iamgabrielma commented:
> PD: I don't think the dark background is a great choice when we have a selected cell in iPad, I'll raise this with design

I've double checked the conversation where we updated them due to color conflicts ( p1751621596213999?thread_ts=1751347733.351649&cid=C070SJRA8DP-slack-C070SJRA8DP ) and it was my miss-understanding: Badge color shouldn't change based on light/dark appearance, we always use the same shade80/shade10 combination. I've updated this again on https://github.com/woocommerce/woocommerce-ios/pull/15887/commits/521dba473940fd0b801e63d353fa62f5250abd22 and now we're good to go 👍 

https://github.com/user-attachments/assets/1a1c3e98-c1d8-4049-b888-5fb03f665ad7

staskus commented:
> I've updated this again on https://github.com/woocommerce/woocommerce-ios/commit/521dba473940fd0b801e63d353fa62f5250abd22 and now we're good to go 👍

Thanks, @iamgabrielma! Good we clarified 👍
